packages/wrangler/src/pages/dev.ts

```diff
@@ -229,99 +236,90 @@ export function Options(yargs: CommonYargsArgv) {
 		});
 }
 
-export const Handler = async ({
-	directory,
-	compatibilityDate,
-	compatibilityFlags,
-	ip,
-	port,
-	inspectorPort,
-	proxy: requestedProxyPort,
-	bundle,
-	noBundle,
-	scriptPath: singleWorkerScriptPath,
-	binding: bindings = [],
-	kv: kvs = [],
-	do: durableObjects = [],
-	d1: d1s = [],
-	r2: r2s = [],
-	ai,
-	service: requestedServices = [],
-	liveReload,
-	localProtocol,
-	httpsKeyPath,
-	httpsCertPath,
-	persistTo,
-	nodeCompat: legacyNodeCompat,
-	experimentalLocal,
-	config: config,
-	_: [_pages, _dev, ...remaining],
-	logLevel,
-	showInteractiveDevSession,
-}: StrictYargsOptionsToInterface<typeof Options>) => {
-	if (logLevel) {
-		logger.loggerLevel = logLevel;
+type PagesDevArguments = StrictYargsOptionsToInterface<typeof Options>;
+
+export const Handler = async (args: PagesDevArguments) => {
+	if (args.logLevel) {
+		logger.loggerLevel = args.logLevel;
 	}
 
-	if (experimentalLocal) {
+	if (args.experimentalLocal) {
 		logger.warn(
 			"--experimental-local is no longer required and will be removed in a future version.\n`wrangler pages dev` now uses the local Cloudflare Workers runtime by default."
 		);
 	}
 
-	if (config) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
 	}
 
-	if (singleWorkerScriptPath !== undefined) {
+	if (args.scriptPath !== undefined) {
 		logger.warn(
 			`\`--script-path\` is deprecated and will be removed in a future version of Wrangler.\nThe Worker script should be named \`_worker.js\` and located in the build output directory of your project (specified with \`wrangler pages dev <directory>\`).`
 		);
 	}
 
-	singleWorkerScriptPath ??= DEFAULT_SCRIPT_PATH;
-
+	const config = readConfig(undefined, args);
+	const resolvedDirectory = args.directory ?? config.pages_build_output_dir;
+	const [_pages, _dev, ...remaining] = args._;
 	const command = remaining;
-
 	let proxyPort: number | undefined;
+	let directory = resolvedDirectory;
+
+	// if we detected a configuration file when attempting to run `pages dev`,
+	// but `pages_build_output_dir` is not specified, we should show a warning,
+	// just in case the user was actually targeting Pages, then proceed with
+	// ignoring the file.
+	if (config.configPath !== "" && directory === undefined && !command.length) {
+		logger.warn(
+			`Found configuration file "${config.configPath}" but missing \`pages_build_output_dir\` field.\n` +
+				`If your configuration file is targeting Pages, please make sure you are using \`wrangler.toml\`, and that the \`pages_build_output_dir\` configuration key is set.`
+		);
+	}
 
 	if (directory !== undefined && command.length > 0) {
 		throw new FatalError(
 			"Specify either a directory OR a proxy command, not both.",
 			1
 		);
 	} else if (directory === undefined) {
-		logger.warn(
```

Comment:
Is there a reason we're removing this deprecation warning?

Reply:
we're not removing it, just moved it inside `spawnProxyProcess`, and ensured we surface it only when `(directory === undefined && (command.length > 0 || port !== undefined))`. Otherwise, it got surfaced in some `pages dev` corner cases, where we'd actually not want it to